lib/_stream_writable.js

```diff
@@ -107,6 +112,13 @@ function WritableState(options, stream) {
 
   // True if the error was already emitted and should not be thrown again
   this.errorEmitted = false;
+
+  // when `state._end()` was called
+  this.flushed = false;
+
+  // NOTE: added here to not pollute the prototype of the WritableStream and to
+  // avoid conflicts with user-land methods
+  this._end = options._end || _end;
```

Comment:
@sonewman I previously did it, and we decided to go with storing it in state for now. To avoid introducing new APIs and breaking existing code that might have been using `_end` for different purposes.

@chrisdickinson If I'll call it `end` - I'll need to write the docs for it :)